cmake/InstallRules.cmake

```diff
@@ -0,0 +1,75 @@
+include(CMakePackageConfigHelpers)
+include(GNUInstallDirs)
+
+set(package ladybird)
+
+install(TARGETS LibAudio LibCore LibFileSystem LibGfx LibIPC LibJS LibWeb LibWebView LibWebSocket LibProtocol LibGUI LibMarkdown LibGemini LibHTTP LibGL LibSoftGPU LibVideo LibWasm LibXML LibIDL LibTextCodec LibCrypto LibLocale LibRegex LibSyntax LibUnicode LibCompress LibTLS LibGLSL LibGPU LibThreading LibSQL
+  EXPORT ladybirdTargets
+  LIBRARY
+    COMPONENT ladybird_Runtime
+    NAMELINK_COMPONENT ladybird_Development
+    DESTINATION ${CMAKE_INSTALL_LIBDIR}
+)
+
+install(TARGETS ladybird WebContent
+  EXPORT ladybirdTargets
+  RUNTIME
+    COMPONENT ladybird_Runtime
+    DESTINATION ${CMAKE_INSTALL_BINDIR}
+  LIBRARY
+    COMPONENT ladybird_Runtime
+    NAMELINK_COMPONENT ladybird_Development
+    DESTINATION ${CMAKE_INSTALL_LIBDIR}
+)
+
+write_basic_package_version_file(
+    "${package}ConfigVersion.cmake"
+    COMPATIBILITY SameMajorVersion
+)
+
+# Allow package maintainers to freely override the path for the configs
+set(
+  ladybird_INSTALL_CMAKEDIR "${CMAKE_INSTALL_DATADIR}/${package}"
+  CACHE PATH "CMake package config location relative to the install prefix"
+)
+mark_as_advanced(ladybird_INSTALL_CMAKEDIR)
+
+install(
+  FILES cmake/LadybirdInstallConfig.cmake
+  DESTINATION "${ladybird_INSTALL_CMAKEDIR}"
+  RENAME "${package}Config.cmake"
+  COMPONENT ladybird_Development
+)
+
+install(
+  FILES "${PROJECT_BINARY_DIR}/${package}ConfigVersion.cmake"
+  DESTINATION "${ladybird_INSTALL_CMAKEDIR}"
+  COMPONENT ladybird_Development
+)
+
+install(
+  EXPORT ladybirdTargets
+  NAMESPACE ladybird::
+  DESTINATION "${ladybird_INSTALL_CMAKEDIR}"
+  COMPONENT ladybird_Development
+)
```

Comment:
This is horrifying, what does all of this do? 😨 All those `ladybird_Runtime` and `ConfigVersion.cmake` and `NAMELINK_COMPONENT`...

Is this even installing Ladybird UI-agnostic parts, or the Ladybird GTK 4?

Reply:
When we say "install(TARGET)", we provide an `EXPORT` which then gets installed here with the `install(EXPORT)` part .

When you install a target, we can install all the associated executables, libraries, archives, and headers all in one go by making sure they're "tied" to the target.

Reply:
The COMPONENT part is so that you can say `cmake --install build --component ladybird_Development` and it will install the files required to build against the project, but if you install `--component ladybird_Runtime` it skips all the non-SDK-like objects and headers.

Reply:
From the install docs, https://cmake.org/cmake/help/latest/command/install.html it says that the NAMELINK_COMPONENT part allows you to separately install your library's actual .so file vs the versioned symlinks in a component install. Apparently debian likes to use this feature.

Reply:
went ahead and yeeted the confusing parts and kept the install rules simple